### ISSUE TITLE
[Feature Request] Muzero and MCTS implementations

### ISSUE THREAD

**Issue by Prakyathkantharaju** [pytorch/rl#1845]
## Motivation

It would be great to have an MCTS and Alphazero implementation, including other model-based RL for benchmarking and comparison. 

## Solution
I can write a loss function of this policy. 

## Alternatives

There are limited RL libraries that have a base implementation of Muzero.
 
## Additional context

None.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
